src/coreclr/debug/ee/rcthread.cpp

```diff
@@ -1365,14 +1365,18 @@ HRESULT DebuggerRCThread::Start(void)
         if (m_thread == NULL)
         {
             LOG((LF_CORDB, LL_EVERYTHING, "DebuggerRCThread failed, err=%d\n", GetLastError()));
-            hr = HRESULT_FROM_GetLastError();
-
+            return HRESULT_FROM_GetLastError();
         }
-        else
+
+        hr = SetThreadDescription(m_thread, W(".NET DebuggerRCThread"));
```

Comment:
This should be done on the thread itself (ie in `DebuggerRCThread::ThreadProcStatic`) so that it works on macOS too.